src/EventProjectorServiceProvider.php

```diff
@@ -60,22 +64,57 @@ public function register(): void
         $this->app->bind(EventSerializer::class, config('event-projector.event_serializer'));
 
         $this->bindCommands();
+
     }
 
     private function bindCommands()
     {
         $this->app->bind('command.event-projector:replay', ReplayCommand::class);
+        $this->app->bind('command.event-projector:cache-event-handlers', CacheEventHandlersCommand::class);
+        $this->app->bind('command.event-projector:clear-event-handlers', ClearCachedEventHandlersCommand::class);
         $this->app->bind('command.make:projector', MakeProjectorCommand::class);
         $this->app->bind('command.make:reactor', MakeReactorCommand::class);
         $this->app->bind('command.make:aggregate', MakeAggregateCommand::class);
         $this->app->bind('command.make:storable-event', MakeStorableEventCommand::class);
 
         $this->commands([
             'command.event-projector:replay',
+            'command.event-projector:cache-event-handlers',
+            'command.event-projector:clear-event-handlers',
             'command.make:projector',
             'command.make:reactor',
             'command.make:aggregate',
             'command.make:storable-event',
         ]);
     }
+
+    private function discoverEventHandlers()
+    {
+        $projectionist = app(Projectionist::class);
+
+        $cachedEventHandlers = $this->getCachedEventHandlers();
```

Comment:
If you avoid `Nullable Return Types`, you could just do:

```php
$projectionist->addEventHandlers($this->getCachedEventHandlers());
```

Reply:
There a subtle difference that happens when `[]` is returned vs `null`. 

If `[]` is returned , it means that the handlers were cached, but there was no handlers were registered. In this case we don't want to autodiscover. 

If `null` is returned, it means that cached did not take place and we should now autoregister.

I could split this up in multiple function, but imho using nullable return types is fine.